packages/go_router/lib/src/route.dart

```diff
@@ -443,6 +453,10 @@ class ShellRoute extends RouteBase {
         assert(route.parentNavigatorKey == null);
       }
     }
+    if (nestedNavigationBuilder != null) {
+      assert(builder == null,
```

Comment:
builder, pageBuilder, nestedNavigationBuilder only one of them should be not null

Reply:
👍 Fixed in PartitionedShellRoute in upcoming commit.